tracer/src/Datadog.Trace/Tagging/HttpTags.cs

```diff
@@ -29,4 +30,35 @@ internal partial class HttpTags : InstrumentationTags, IHasStatusCode
         [Tag(Trace.Tags.HttpStatusCode)]
         public string HttpStatusCode { get; set; }
     }
+
+    internal partial class HttpV1Tags : HttpTags
+    {
+        private string _peerServiceOverride = null;
+        private string _host = null;
+
+        // Use a private setter for setting the "peer.service" tag so we avoid
+        // accidentally setting the value ourselves and instead calculate the
+        // value from predefined precursor attributes.
+        // However, this can still be set from ITags.SetTag so the user can
+        // customize the value if they wish.
+        [Tag(Trace.Tags.PeerService)]
+        public string PeerService
+        {
+            get => _peerServiceOverride ?? _host;
+            private set => _peerServiceOverride = value;
+        }
+
+        [Tag(Trace.Tags.PeerServiceSource)]
+        public string PeerServiceSource
+        {
+            get
+            {
+                return _peerServiceOverride is not null
+                        ? "peer.service"
+                        : "network.destination.name";
+            }
+        }
+
+        public void SetHost(string host) => _host = host;
```

Comment:
This feels wrong, but I was avoiding adding a new tag since we already encode the HTTP url as a tag. And our strict validation doesn't allow us to use properties unless they have a TagAttribute or MetricAttribute attached to it.

Is this implementation fine? Or should we just add a new tag like `out.host` (similar to what MongoDbTags/SqlTags) which will allow us to easily use a property?

Reply:
I think I've seen that `out.host` tag and `Host` property in a decent number of the tags classes - maybe it'd be best to standardize on doing that versus this way?

But I think that this `SetHost` is fine since it seems that we already have the information and it avoids us adding a new semi-redundant tag